packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -187,58 +211,73 @@ public ResponseEntity<TaskResponse> createConfigureModelTask(
 		final Schema.Permission permission =
 				projectService.checkPermissionCanRead(currentUserService.get().getId(), projectId);
 
-		try {
+		// Grab the document
+		final Optional<DocumentAsset> document = documentAssetService.getAsset(documentId, permission);
+		if (document.isEmpty()) {
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("document.not-found"));
+		}
 
-			// Grab the document
-			final Optional<DocumentAsset> document = documentAssetService.getAsset(documentId, permission);
-			if (document.isEmpty()) {
-				throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Document not found");
-			}
+		// make sure there is text in the document
+		if (document.get().getText() == null || document.get().getText().isEmpty()) {
+			log.warn("Document {} has no text to send", documentId);
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("document.extraction.not-done"));
```

Comment:
I know that this was a `404` before, but technically is that right? I think that this may [technically](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/424) be a `424`.

Reply:
Do you mean `422 Unprocessable Entity`? I don't see how a `424 Failed Dependency` explain the issue.

Reply:
"Regular web servers will normally not return this status code." this is from the link you sent

Reply:
Are you calling HMI-Server just a regular web server ;)?